ide-common/src/main/java/org/digma/intellij/plugin/view/ListGroupManager.java

```diff
@@ -5,14 +5,20 @@
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Optional;
 import java.util.function.Supplier;
 
 public class ListGroupManager {
 
     private final Map<String, GroupListViewItem> groups = new HashMap<>();
 
     public GroupListViewItem getOrCreateGroup(String groupKey, Supplier<GroupListViewItem> supplier) {
-        return groups.computeIfAbsent(groupKey, key -> supplier.get());
+        Optional<String> existingMatchingGroupKey = groups.keySet()
```

Comment:
Great catch! thanks!
I've just committed the fix for this. Now it works correct for sure because now I save into groups map only ShortHttpRouteName like this ("POST Transfer/TransferFunds").
`public class ListGroupManager {

    private final Map<String, GroupListViewItem> groups = new HashMap<>();

    public GroupListViewItem getOrCreateGroup(String groupKey, Supplier<GroupListViewItem> supplier) {
        String shortRouteName = getShortHttpRouteName(groupKey);
        return groups.computeIfAbsent(shortRouteName, key -> supplier.get());
    }

    public Collection<GroupListViewItem> getGroupItems() {
        return groups.values();
    }

    private String getShortHttpRouteName(String fullRouteName) {
        if (fullRouteName.startsWith(HTTP_SCHEMA)) {
            return fullRouteName.replace(HTTP_SCHEMA + ":", "");
        }
        if (fullRouteName.startsWith("HTTP ")) {
            return fullRouteName.replace("HTTP ", "");
        }
        else return fullRouteName;
    }

}`